File: Week1/김다빈/BOJ_18258/18258.py
from collections import deque
import sys

n = int(input())
dec = deque()


def push(x):
    dec.append(x)


def pop():
    if dec:
        print(dec.popleft())
    else:
        print(-1)


def size():
    print(len(dec))


def empty():
    if dec:
        print(0)
    else:
        print(1)


def front():
    if dec:
        print(dec[0])
    else:
        print(-1)


def back():
    if dec:
        print(dec[-1])
    else:
        print(-1)


for _ in range(n):
    # n의 값이 크기 때문에 input()으로 입력받으면 시간초과가 난다.
    cmd = list(sys.stdin.readline().split())

    if cmd[0] == "push":
        push(int(cmd[1]))
    elif cmd[0] == "pop":
        pop()
    elif cmd[0] == "size":
        size()
    elif cmd[0] == "empty":
        empty()
    elif cmd[0] == "front":
        front()
    else:
        back()

File: Week1/김다빈/BOJ_18258/18258.java
import java.io.*;
import java.util.StringTokenizer;
import java.util.LinkedList;
import java.util.Deque;


public class Main {
    private static Deque<Integer> queue = new LinkedList<>();
    private static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));

    private static void push(int x) {
        queue.addLast(x);
    }

    private static void pop() throws Exception {
        if (queue.isEmpty()) {
            bw.write(-1 + "\n");
        } else {
            bw.write(queue.pollFirst() + "\n");
        }
    }

    private static void size() throws Exception {
        bw.write(queue.size() + "\n");
    }

    private static void empty() throws Exception {
        boolean result = queue.isEmpty();

        if (result) {
            bw.write(1 + "\n");
        } else {
            bw.write(0 + "\n");
        }
    }


    private static void front() throws Exception {
        if (queue.isEmpty()) {
            bw.write(-1 + "\n");

        } else {
            bw.write(queue.peekFirst() + "\n");
        }
    }

    private static void back() throws Exception {
        if (queue.isEmpty()) {
            bw.write(-1 + "\n");

        } else {
            bw.write(queue.peekLast() + "\n");
        }
    }

    public static void main(String[] args) throws Exception {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        int n = Integer.parseInt(br.readLine());

        for (int i = 0; i < n; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String cmd = st.nextToken();

            switch (cmd) {
                case "push":
                    push(Integer.parseInt(st.nextToken()));
                    break;

                case "pop":
                    pop();
                    break;

                case "size":
                    size();
                    break;

                case "empty":
                    empty();
                    break;

                case "front":
                    front();
                    break;

                case "back":
                    back();
                    break;
            }
        }
        bw.flush();
        bw.close();
    }
}